packages/flame/lib/src/components/isometric_tile_map_component.dart

```diff
@@ -32,16 +32,25 @@ class IsometricTileMapComponent extends PositionComponent {
   /// Optionally provide a new tile size to render it scaled.
   Vector2? destTileSize;
 
+  /// This is the vertical height of each block in the tile set.
+  ///
+  /// Note: this must be measured in the destination space.
+  double? tileHeight;
+
   IsometricTileMapComponent(
     this.tileset,
     this.matrix, {
     this.destTileSize,
+    this.tileHeight,
     Vector2? position,
   }) : super(position: position);
 
   /// This is the size the tiles will be drawn (either original or overwritten).
   Vector2 get effectiveTileSize => destTileSize ?? tileset.srcSize;
 
+  /// This is the vertical height of each block; by default it's half the tile size.
+  double get effectiveTileHeight => tileHeight ?? (effectiveTileSize.x / 2);
```

Comment:
Should this be a late final so the value gets memoized?

Reply:
it can't, because the values are not final

Reply:
That seems odd to me, should these values be changeable, I mean, once you set the tileHeight it should remain the same for the entire existence of the component. Or am I missing something?

Reply:
I don't see why, if you change it everything will work with the new size. but I can make it final as well to get that performance boost, no strong opinion here

Reply:
Perfomance boost is exactly why I am suggesting it hahhaa

Reply:
even if I make it final it's still unsafe because the user can change the contents of the vector2, so this will require a small api change. also I don't think performance is important here because a tilemap should not be rendered every frame anyway. since this is existing code, I would prefer to leave as is for now, we can followup later.

Reply:
SGTM